src/org/wikipedia/Wiki.java

```diff
@@ -2251,7 +2250,8 @@ public void purge(boolean links, String... titles) throws IOException
             url.append(encode(template, false));
         }
         url.append("&titles=");
-        
+
+        String[] titlestrings = constructTitleString(url.length() + "&tlcontinue=".length(), titles, true);
```

Comment:
I am not sure how long the `tlcontinue` value can be; Apparently it can contain a full title, so I made it 1000, to err on the safe side.